sklearn/decomposition/tests/test_fastica.py

```diff
@@ -92,8 +92,12 @@ def test_fastica_simple(add_noise, global_random_seed, global_dtype):
     mixing = mixing.astype(global_dtype)
     m = np.dot(mixing, s)
 
+    # XXX we generate noise even when add_noise=False to avoid a rare case
+    # where the test fails for float32. For more details see:
+    # https://github.com/scikit-learn/scikit-learn/issues/24131#issuecomment-1208091119
```

Comment:
```suggestion
    # XXX we consume the RNG even when add_noise=False to avoid a rare case
    # where the test fails for float32 with (SciPy or LAPACK bug?):
    # ValueError: On entry to SLASCL parameter number 4 had an illegal value.
    # For more details see:
    # https://github.com/scikit-learn/scikit-learn/issues/24131#issuecomment-1208091119
```